source/extensions/quic_listeners/quiche/active_quic_listener.cc

```diff
@@ -112,14 +115,30 @@ void ActiveQuicListener::onData(Network::UdpRecvData& data) {
                                   /*packet_headers=*/nullptr, /*headers_length=*/0,
                                   /*owns_header_buffer*/ false);
   quic_dispatcher_->ProcessPacket(self_address, peer_address, packet);
+
+  // On event loop runs where this listener reads or has packets posted to it, process buffered
+  // packets on the next loop. This ensures that the limit of processed client hellos per loop is
+  // honored, and that buffered client hellos are processed in a timely fashion.
+  udp_listener_->activateRead();
```

Comment:
I don't know the full flow, but shouldn't this be gated by `HasChlosBuffered` like below? Stepping back, it seems like the read ready event already handles this re-fire so why is this needed here?

Reply:
This is needed for two cases:

1) `onReadReady()` is called at the beginning of the event loop, before packets are processed. Processing packets can add a buffered chlo.

2) If another worker posts a packet to this one, `onReadReady()` may not be called during this event loop (depending on incoming network traffic and how the kernel shards it). 

I'll elaborate on the code-comment to clarify this.

Reply:
Oh, I mis-read your question.

Yes, it probably should.